packages/amplify-appsync-simulator/src/data-loader/http/index.ts

```diff
@@ -0,0 +1,70 @@
+import axios, { AxiosResponse } from 'axios';
+import { AmplifyAppSyncSimulatorDataLoader } from '..';
+import { AppSyncSimulatorDataSourceHttpConfig } from '../../type-definition';
+
+export class HttpDataLoader implements AmplifyAppSyncSimulatorDataLoader {
+  private endpoint: string;
+  private env: string;
+  private region: string;
+
+  constructor(config: AppSyncSimulatorDataSourceHttpConfig) {
+    this.endpoint = config?.httpConfig?.endpoint;
+    this.env = config?.httpConfig?.env === '${env}' ? 'NONE' : config?.httpConfig?.env;
+    this.region = config?.httpConfig?.region;
+  }
+
+  public async load(payload: any): Promise<any> {
+    const reqEndpoint = this.endpoint + this.parseUrl(payload.resourcePath, this.env, this.region);
+    try {
+      const axiosRes: AxiosResponse = await axios({
+        method: payload.method,
+        url: reqEndpoint,
+        headers: payload.params.headers,
+        data: { query: payload.params.query },
+        params: { query: payload.params.query },
+      });
+      const result = this.parseResponse(axiosRes);
+      const cfxResult = { body: result, statusCode: axiosRes.status, headers: axiosRes.headers };
+      return cfxResult;
+    } catch (error) {
+      console.log('HTTP Data source failed with the following error:');
+      console.error(error);
+      if (error.response.status === 400) console.error(error.response.data);
+      throw error;
+    }
+  }
+
+  private parseResponse = (response: AxiosResponse) => {
+    let result = response.data?.data;
+    if (!result) {
+      throw new Error('Missing result in response (response.data.data is undefined.)');
+    }
+    let first = Object.keys(result)[0];
+
+    if (Array.isArray(result[first])) result = result[first];
+    else result = this.flattenObject(result);
```

Comment:
That `response.data.data` returns an object like this:
```js
{
  getPost: { id: "1", title: "A Post" }
}
// or
{
  listPosts: [
    {
      id: "1",
      title: "A Post"
    },
  // ...
]
```

but the resolver expects a one-level depth object, so a single Post object returned would look like this after being flattened:
```js
{
  id: "1",
  title: "A Post"
}
```
but if an array of objects is being returned, we flatten down to the array itself (`result = result[first]`) so that the returned object looks like this:
```js
[
  {
    id: "1",
    title: "A Post"
  },
  // ...
]
```
Does that clear it up? I'll also add a comment in the code